support/Python/DirectoryStructure.py

```diff
@@ -126,3 +126,99 @@ def list_segments(segments_dir: Union[str, Path]) -> List[Segment]:
         return []
     matches = map(Segment.match, segments_dir.iterdir())
     return sorted(match for match in matches if match)
+
+
+@dataclass(frozen=True, order=True)
+class PipelineStep:
+    """A complete simulation iteration
+
+    Represents a full simulation run. Multiple iterations can exist,
+    organized in a directory structure like this:
+
+    ```
+    BASE_DIR/
+        001_InitialData/
+            Segments/
```

Comment:
Initial data doesn't use segments or checkpoints, so list `InputFile.yaml` and `Submit.sh` directly in the pipeline step directory for this example.